src/lsp/lp_doc.ml

```diff
@@ -12,6 +12,106 @@
 
 open Core
 module LSP = Lsp_base
+(*
+exception NoPosition of string *)
+
+module type RangeMap = sig
+
+  type point
+  type interval
+  type 'a t
+
+  val create_point : int -> int -> point
+  val l : point -> int
+  val c : point -> int
+
+  val interval_start : interval -> point
+  val interval_end : interval -> point
+
+  val create_interval : point -> point -> interval
+
+  type cmp = Before | In | After
+
+  val in_range : point -> interval -> cmp
+
+  val add : interval -> 'a -> 'a t -> 'a t
+  val find : point -> 'a t -> 'a option * interval
+
+  val empty : 'a t
+
+  val rangemap_to_string : ('a * string) t -> string
```

Comment:
What is the use of this function? If it is for debug, isn't it better to use Format?

Reply:
I'm not sure how I should use Format here. By providing a string version of the map I just need to type LIO.log_error str, but if Format is more practical I'll change that.

Reply:
That's for debug, it could be further tweaked indeed, but not a blocker iMHO.